10-JavaScript/js-intro.md

```diff
@@ -0,0 +1,439 @@
+1. `4 / 2;`
+1. `4 / 2.0;`
+1. `4.0 + 2.0`
+1. `4 + 2.1`
+1. `Infinity;`
+1. `+Infinity;`
+1. `-Infinity;`
+1. `Number.MAX_VALUE;`
+1. `Infinity > Number.MAX_VALUE;`
+1. `0.1 + 0.2 == 0.3;`
+1. `0.1 + 0.2;`
+1. `'asdf' - 5;`
+
+<details>
+
+  <summary>
+    Compare your answers here
+  </summary>
+
+  1. `4`
+  1. `2`
+  1. `2`
+  1. `2`
+  1. `6`
+  1. `6.1`
+  1. `Infinity`
+  1. `Infinity`
+  1. `-Infinity`
+  1. `1.7976931348623157e+308`
+  1. `true`
+  1. `false`
+  1. `0.30000000000000004`
+  1. `NaN`
+
+  We will not spend the time to go into the specifics and the "why"s of JavaScript at the moment; for now, be satisfied with the mystery, curiosity, and difference you can see.
+
+</details>
+
+### `Strings` are declared with `""`, `''`, or backticks `    ``    `.
+
+The idea of Strings is the same between JavaScript and Ruby. However, these are the following differences we will highlight:
+
+- In JavaScript, strings declared with double-quotes and strings declared with single-quotes are the same
+- In those cases, we will standardize on using single-quotes `''`
+- **We can only use backticks for string interpolation**
+- The syntax for string interpolation is with `${}`
+- JavaScript does not have symbols, and uses strings wherever Ruby would use symbols
+
+In your Node REPL, input these lines. Do you get what you expect?
+
+1. `"Hello World!";`
+1. `'Hello World!';`
+1. `'Hello World!'.length`;
+1. `'A triangle has ${ 2 + 1 } sides.';`
+1. `` `A triangle has ${ 2 + 1 } sides` ``
+
+<details>
+
+  <summary>
+    Compare your answers here
+  </summary>
+
+  1. `Hello World!`
+  1. `Hello World!`
+  1. `12`
+  1. `'A triangle has ${ 2 + 1 } sides.'`
+  1. `'A triangle has 3 sides'`
+
+</details>
+
+### Arrays
+
+[Arrays](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array) are similar to Ruby arrays. They are declared and accessed with square brackets ([]).
+
+In your Node REPL, input these lines. Do you get what you expect?
+
+1. `[1, 2, 3, 4];`
+1. `['foo', 1, 2, 'bar', 'baz'];`
+1. `['foo', 1, 2, 'bar', 'baz'][0];`
+1. `['foo', 1, 2, 'bar', 'baz'][3];`
+1. `['foo', 1, 2, 'bar', 'baz'][-1];`
+1. `['foo', 1, 2, 'bar', 'baz'].length;`
+1. `['foo', ['this', 'is', 'a', 'nested', 'array'], 1, 2];`
+1. `['foo', ['this', 'is', 'a', 'nested', 'array'], 1, 2].length;`
+
+<details>
+
+  <summary>
+    Compare your answers here
+  </summary>
+
+  1. `[ 1, 2, 3, 4 ]`
+  1. `[ 'foo', 1, 2, 'bar', 'baz' ]`
+  1. `'foo'`
+  1. `'bar'`
+  1. `undefined` (we will get into what this means in a minute, but know that it means that we can't index by `-1` in JavaScript)
+  1. `5`
+  1. `[ 'foo', [ 'this', 'is', 'a', 'nested', 'array' ], 1, 2 ]`
+  1. `4`
+
+</details>
+
+### `null` represents an empty value
+
+Similar to Ruby's `nil`, JavaScript's `null` represents an empty value.
+
+In your Node REPL, input these lines. Do you get what you expect?
+
+1. `null;`
+1. `null == true;`
+1. `null == false;`
+1. `null.length;`
```

Comment:
I know we haven't touched conditionals yet, but maybe it would be worthwhile to add something like

```js
if (null) {
  console.log("null is truthy");
} else {
  console.log("null is falsey");
}
```

IDK